### PR TITLE
Session3

### PR DESCRIPTION
**Pull request by emorejejeh**

### CODE REVIEW

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        public ItemsController(IItemService itemService)
+        {
+            _itemService = itemService;
+        }
+        public int Get(int id)
+        {
+            return id;
+        }
+
+        public void Post(ItemCreateApiModel model)
```
Comment (jegtugado) [inline ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: we need to use http verb Post in order to post data to the API. Please use [HttpPost]. And indicate the binding source such as [FromBody] as discussed during session 3

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            _itemService.Save(new ItemDto { Item = model.Item });
```
Comment (jegtugado) [inline ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: We need to check for model validation result using ModelState.IsValid to react to validation errors

File: DomainModels/Item.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace DomainModels
+{
+    public class Item
+    {
```
Comment (jegtugado) [inline DomainModels/Item.cs]: Sir please add a public string property Item like in your Item DTO model

File: Services/ItemService.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using Services.DTO;
+using Services.Interfaces;
+
+namespace Services
+{
+    public class ItemService : IItemService
+    {
+        public void Save(ItemDto dto)
+        {
```
Comment (jegtugado) [inline Services/ItemService.cs:10]: Need to map ItemDTO into an Item domain model object

**Review by jegtugado — CHANGES_REQUESTED** [review]

**Comment by jegtugado** [comment]
Hi sir. A little adjustment is needed for the PR. Good job resolving the gitignore issue!

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs (hunk 1 above)
Comment (emorejejeh) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: updated

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs (hunk 2 above)
Comment (emorejejeh) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: updated

File: Services/ItemService.cs (hunk 4 above)
Comment (emorejejeh) [reply Services/ItemService.cs:10]: updated

**Review by jegtugado — APPROVED** [review]

**Pull request merged**